### PR TITLE
Parodos POC

### PR DESCRIPTION
**Pull request by fjtirado**
This example illustrate how to use subflowRef and event state to to achieve shared context functionality. The initial setup workflow establish the parameters for the other workflows sharing the context, which are instantiated through events.

### CODE REVIEW

**Comment by kie-ci3** [comment]
**PR job** `#1` was: **UNSTABLE**
Possible explanation: This should be test failures


<details>
<summary><b>Reproducer</b></summary>



build-chain build full_downstream  -f 'https://raw.githubusercontent.com/${AUTHOR:apache}/incubator-kie-kogito-pipelines/${BRANCH:main}/.ci/buildchain-config-pr-cdb.yaml' -o 'bc' -p apache/incubator-kie-kogito-examples -u https://github.com/apache/incubator-kie-kogito-examples/pull/1918 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-examples-pr/job/PR-1918/1/display/redirect

**Test results:**
- PASSED: 383
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-examples-pr/job/PR-1918/1/testReport/org.acme/AcmeExchangeResourceIT/unauthorizedTest/">org.acme.AcmeExchangeResourceIT.unauthorizedTest</a></summary>
java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors<br/> [error]: Build step io.quarkus.oidc.deployment.devservices.keycloak.KeycloakDevServicesProcessor#startKeycloakContainer threw an exception: java.lang.RuntimeException: org.testcontainers.containers.ContainerLaunchException: Container startup failed for image quay.io/keycloak/keycloak:23.0.7<br/> at io.quarkus.oidc.deployment.devservices.keycloak.KeycloakDevServicesProcessor.startKeycloakContainer(KeycloakDevServicesProcessor.java:249)<br/> at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)<br/> at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)<br/> at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)<br/> at java.base/java.lang.reflect.Method.invoke(Method.java:568)<br/> at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:849)<br/> at io.quarkus.builder.BuildContext.run(BuildContext.java:256)<br/> at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)<br/> at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)<br/> at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)<br/> at java.base/java.lang.Thread.run(Thread.java:833)<br/> at org.jboss.threads.JBossThread.run(JBossThread.java:501)<br/>Caused by: org.testcontainers.containers.ContainerLaunchException: Container startup failed for image quay.io/keycloak/keycloak:23.0.7<br/> at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:359)<br/> at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:330)<br/> at io.quarkus.oidc.deployment.devservices.keycloak.KeycloakDevServicesProcessor.lambda$startContainer$4(KeycloakDevServicesProcessor.java:376)<br/> at java.base/java.util.Optional.orElseGet(Optional.java:364)<br/> at io.quarkus.oidc.deployment.devservices.keycloak.KeycloakDevServicesProcessor.startContainer(KeycloakDevServicesProcessor.java:401)<br/> at io.quarkus.oidc.deployment.devservices.keycloak.KeycloakDevServicesProcessor.startKeycloakContainer(KeycloakDevServicesProcessor.java:197)<br/> ... 11 more<br/>Caused by: org.rnorth.ducttape.RetryCountExceededException: Retry limit hit with exception<br/> at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:88)<br/> at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:344)<br/> ... 16 more<br/>Caused by: org.testcontainers.containers.ContainerLaunchException: Could not create/start container<br/> at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:563)<br/> at org.testcontainers.containers.GenericContainer.lambda$doStart$0(GenericContainer.java:354)<br/> at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:81)<br/> ... 17 more<br/>Caused by: org.testcontainers.containers.ContainerLaunchException: Timed out waiting for log output matching '.*Keycloak.*started.*'<br/> at org.testcontainers.containers.wait.strategy.LogMessageWaitStrategy.waitUntilReady(LogMessageWaitStrategy.java:47)<br/> at org.testcontainers.containers.wait.strategy.AbstractWaitStrategy.waitUntilReady(AbstractWaitStrategy.java:52)<br/> at org.testcontainers.containers.GenericContainer.waitUntilContainerStarted(GenericContainer.java:909)<br/> at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:500)<br/> ... 19 more<br/>
</details>

**Comment by kie-ci3** [comment]
**PR job** `#2` was: **UNSTABLE**
Possible explanation: This should be test failures


<details>
<summary><b>Reproducer</b></summary>



build-chain build full_downstream  -f 'https://raw.githubusercontent.com/${AUTHOR:apache}/incubator-kie-kogito-pipelines/${BRANCH:main}/.ci/buildchain-config-pr-cdb.yaml' -o 'bc' -p apache/incubator-kie-kogito-examples -u https://github.com/apache/incubator-kie-kogito-examples/pull/1918 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-examples-pr/job/PR-1918/2/display/redirect

**Test results:**
- PASSED: 383
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-examples-pr/job/PR-1918/2/testReport/org.acme/AcmeExchangeResourceIT/unauthorizedTest/">org.acme.AcmeExchangeResourceIT.unauthorizedTest</a></summary>
java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors<br/> [error]: Build step io.quarkus.oidc.deployment.devservices.keycloak.KeycloakDevServicesProcessor#startKeycloakContainer threw an exception: java.lang.RuntimeException: org.testcontainers.containers.ContainerLaunchException: Container startup failed for image quay.io/keycloak/keycloak:23.0.7<br/> at io.quarkus.oidc.deployment.devservices.keycloak.KeycloakDevServicesProcessor.startKeycloakContainer(KeycloakDevServicesProcessor.java:249)<br/> at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)<br/> at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)<br/> at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)<br/> at java.base/java.lang.reflect.Method.invoke(Method.java:568)<br/> at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:849)<br/> at io.quarkus.builder.BuildContext.run(BuildContext.java:256)<br/> at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)<br/> at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)<br/> at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)<br/> at java.base/java.lang.Thread.run(Thread.java:833)<br/> at org.jboss.threads.JBossThread.run(JBossThread.java:501)<br/>Caused by: org.testcontainers.containers.ContainerLaunchException: Container startup failed for image quay.io/keycloak/keycloak:23.0.7<br/> at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:359)<br/> at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:330)<br/> at io.quarkus.oidc.deployment.devservices.keycloak.KeycloakDevServicesProcessor.lambda$startContainer$4(KeycloakDevServicesProcessor.java:376)<br/> at java.base/java.util.Optional.orElseGet(Optional.java:364)<br/> at io.quarkus.oidc.deployment.devservices.keycloak.KeycloakDevServicesProcessor.startContainer(KeycloakDevServicesProcessor.java:401)<br/> at io.quarkus.oidc.deployment.devservices.keycloak.KeycloakDevServicesProcessor.startKeycloakContainer(KeycloakDevServicesProcessor.java:197)<br/> ... 11 more<br/>Caused by: org.rnorth.ducttape.RetryCountExceededException: Retry limit hit with exception<br/> at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:88)<br/> at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:344)<br/> ... 16 more<br/>Caused by: org.testcontainers.containers.ContainerLaunchException: Could not create/start container<br/> at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:563)<br/> at org.testcontainers.containers.GenericContainer.lambda$doStart$0(GenericContainer.java:354)<br/> at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:81)<br/> ... 17 more<br/>Caused by: org.testcontainers.containers.ContainerLaunchException: Timed out waiting for log output matching '.*Keycloak.*started.*'<br/> at org.testcontainers.containers.wait.strategy.LogMessageWaitStrategy.waitUntilReady(LogMessageWaitStrategy.java:47)<br/> at org.testcontainers.containers.wait.strategy.AbstractWaitStrategy.waitUntilReady(AbstractWaitStrategy.java:52)<br/> at org.testcontainers.containers.GenericContainer.waitUntilContainerStarted(GenericContainer.java:909)<br/> at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:500)<br/> ... 19 more<br/>
</details>

File: serverless-workflow-examples/serverless-workflow-subflows-event/README.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
+# serverless-workflow-subflow-events
+
+This example illustrate how to trigger workflows manually with additional parameters calculated by an initial workflow. 
+The workflow that initially setup the parameters is executed as start state.
```
Comment (gmunozfe) [inline serverless-workflow-examples/serverless-workflow-subflows-event/README.md]: Possibly rephrase:
```suggestion
The workflow responsible for setting up the parameters is executed as the start state.
```

File: serverless-workflow-examples/serverless-workflow-subflows-event/README.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+```
+curl --location 'http://localhost:8080/executeA' \
+--header 'Content-Type: application/json' \
+--data '{
+    "id" : "1",
+   "specversion" : "1.0",
+   "type" : "executeA",
+   "source" : "manual", 
+  "data" : {
```
Comment (gmunozfe) [inline serverless-workflow-examples/serverless-workflow-subflows-event/README.md]: Check indentation

File: serverless-workflow-examples/serverless-workflow-subflows-event/src/main/resources/application.properties [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+quarkus.devservices.enabled=false
+#
```
Comment (gmunozfe) [inline serverless-workflow-examples/serverless-workflow-subflows-event/src/main/resources/application.properties]: you can remove it

**Review by gmunozfe — APPROVED** [review]
Looks good to me, fantastic example @fjtirado !

File: serverless-workflow-examples/serverless-workflow-subflows-event/README.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+```
+
+We need to write down the id returned by the previous steps and invoke `workflowA` through a cloud event containing that id as `kogitoprocrefid` attibute. 
+
+```
+curl --location 'http://localhost:8080/executeA' \
+--header 'Content-Type: application/json' \
+--data '{
+   "id" : "1",
+   "specversion" : "1.0",
```
Comment (ricardozanini) [inline serverless-workflow-examples/serverless-workflow-subflows-event/README.md:37]:
```suggestion
   "specversion" : "0.8",
```

File: serverless-workflow-examples/serverless-workflow-subflows-event/README.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+```
+
+The main workflow is still active, waiting for execution of workflow B. Lets execute it sending another cloud event. 
+
+```
+curl --location 'http://localhost:8080/executeB' \
+--header 'Content-Type: application/json' \
+--data '{
+    "id": "1",
+    "specversion": "1.0",
```
Comment (ricardozanini) [inline serverless-workflow-examples/serverless-workflow-subflows-event/README.md:63]:
```suggestion
    "specversion": "0.8",
```

**Review by ricardozanini — APPROVED** [review]

File: serverless-workflow-examples/serverless-workflow-subflows-event/README.md (hunk 5 above)
Comment (fjtirado) [reply serverless-workflow-examples/serverless-workflow-subflows-event/README.md:63]: @ricardozanini This is the CloudEvent version, not the SWF one ;)

**Pull request merged**